src/carousel/carousel.ts

```diff
@@ -196,6 +212,48 @@ export class NgbCarousel implements AfterContentChecked,
     return isFirstSlide ? (this.wrap ? slideArr[slideArr.length - 1].id : slideArr[0].id) :
                           slideArr[currentSlideIdx - 1].id;
   }
+
+  private _getSlideEventDirection(currentActiveSlideId: string, nextActiveSlideId: string):
+      NgbSlideCarouselEventDirection {
+    const currentActiveSlideIdx = this._getSlideIdxById(currentActiveSlideId);
+    const nextActiveSlideIdx = this._getSlideIdxById(nextActiveSlideId);
+
+    return currentActiveSlideIdx > nextActiveSlideIdx ? NgbSlideCarouselEventDirection.RIGHT :
```

Comment:
I would expect that when you are on the first slide and click the back arrow that the direction would be RIGHT. However, when you are on the first slide with this logic it will always return LEFT, regardless of which direction you are moving in. The same thing applies if you are on the last slide, the direction will always be RIGHT, regardless of which direction you move in.